docs/inputs.md

```diff
@@ -0,0 +1,319 @@
+- name: meaning_of_life
+  type: Numeric
+  value: 42
+  required: true
+  priority: 70
+```
+
+All [input options](#input-options-reference) are supported in metadata files.
+
+There are two major advantages to defining inputs in profile metadata:
+ 1. The inputs and their configuration are listed explicitly in simple YAML in one place - a consumer of your profile does not need to read through the control code to find the inputs.
+ 2. You can set inputs in other profiles that you depend on using profile inheritance.
+
+### Using inputs with Profile inheritance
+
+When your profile relies on another profile using the `depends` key in the metadata file, you can set (that is, override) the value of the input in the dependent profile by including the `profile` option and naming the dependent profile.
+
+
+```yaml
+# child inspec.yml
+name: child
+inputs:
+- name: favorite_food
+  value: pizza
+```
+
+```yaml
+# wrapper inspec.yml
+name: wrapper
+depends:
+- name: child
+  path: ../child
+inputs:
+- name: favorite_food
+  value: broccoli
+  profile: child       # <----- REQUIRED to override the value in InSpec 4
+```
+
+In Chef InSpec 4+, every Input is namespaced: so you could have an input named `wrapper/favorite_food` and one named `child/favorite_food`. Within the `wrapper` profile metadata file, if no explicit profile option is set, `wrapper` is assumed to be the profile.
+
+## Setting Input values using `--input-file`
+
+You may also provide inputs and values via YAML files on the command line. The format is simple:
+
+```yaml
+an_input: a_value
+another_input: another_value
+```
+
+CLI-set Inputs have a priority of 40.
+
+As of Chef InSpec 4.3.2, this mechanism has the following limitations:
+
+  1. No [input options](#input-options-reference) may be set - only the name and value
+  2. Because the CLI is outside the scope of any one profile and the inputs don't take options, the Inputs are clumsily copied into every profile, effectively making the CLI mechanism global.
+
+## Input Options Reference
+
+### Name
+
+Required `String`. This identifies the Input.
+
+Allowed in: All. When used in DSL and Metadata, the name is unique with the current profile; when used in CLI input files, audit cookbook, and kitchen-inspec, the input is copied across all profiles using the same name.
+
+### Description
+
+Optional `String`. Human-meaningful explanation of the Input.
+
+Allowed in: DSL, Metadata
+
+### Value
+
+Optional, any Ruby or YAML type. This is the value that will be available when you read the input. See [Reading Inputs](#reading-inputs-in-control-dsl).
+
+Allowed in: All
+
+### Type
+
+Optional, `String`, one of `String`, `Numeric`, `Regexp`, `Array`, `Hash`, `Boolean`, or `Any`. If provided, the value with be checked to see if it is of the corresponding type. Note that `Regexp` indicates that the value itself should be a regular expression, not that it should match any particular one.
```

Comment:
The first section of information is a bit confusing. Can you please clarify about "Optional, `String`, one of `String`, `Numeric`, `Regexp`, `Array`, `Hash`, `Boolean`, or `Any`"? 

Is it that the type value is limited to `String`, `Numeric`, `Regexp`, `Array`, `Hash`, `Boolean`, or `Any`? More words may be needed to explain.